### ISSUE TITLE
undefined method `name` while accessing to `RuboCop::Cop::Offense::highlighted_area`

### ISSUE THREAD

**Issue by snutij** [rubocop/rubocop#12527]
Hello 👋  I'm trying to access to the `highlighted_area` method of an offense, which is on the public api, but it's raising an error.

--------

## Expected behavior

Should return a `Parser::Source::Range` as documented here: 

https://github.com/rubocop/rubocop/blob/b1bcb3134d49a60061d08ef168ae996d70d8ce5c/lib/rubocop/cop/offense.rb#L138-L144

## Actual behavior

If I'm adding a debug in the `clang_formatter`:
https://github.com/rubocop/rubocop/blob/b1bcb3134d49a60061d08ef168ae996d70d8ce5c/lib/rubocop/formatter/clang_style_formatter.rb#L51-L56

Accessing to `highlighted_area` raise a `NoMethodError`

```ruby
(ruby) highlighted_area
#<NoMethodError: undefined method `name' for "  def dummy(node)":String> rescued during inspection
(ruby) highlighted_area.source_buffer
"  def dummy(node)"
(ruby) highlighted_area.begin_pos
2
(ruby) highlighted_area.source
"def dummy(node)"
``` 

## Steps to reproduce the problem

- Create a file `foo.rb`

```ruby
class Foo
  def dummy(node)
    puts "foo"
    puts "foo"
    puts "foo"
  end
end
``` 

- Enable the rule `Metrics/MethodLength` with `Max: 2`
- Add a breakpoint into `lib/rubocop/formatter/clang_style_formatter.rb`
- Run `bin/rubocop -d --only Metrics/MethodLength foo.rb`

## RuboCop version

```
bundle exec rubocop -V
1.58.0 (using Parser 3.2.2.4, rubocop-ast 1.30.0, running on ruby 3.2.2) [arm64-darwin23]
  - rubocop-minitest 0.33.0
  - rubocop-rake 0.6.0
  - rubocop-sorbet 0.7.5
```

**Comment by snutij** [comment]
For the record, the first argument to [initialize a range](https://github.com/whitequark/parser/blob/7340bc3d1a056467612cdc95d306e32bcc2cce45/lib/parser/source/range.rb#L37) should be a [`Parser::Source::Buffer`](https://github.com/whitequark/parser/blob/master/lib/parser/source/buffer.rb), which is not the case, as this is [initialized with `source_line`](https://github.com/rubocop/rubocop/blob/b1bcb3134d49a60061d08ef168ae996d70d8ce5c/lib/rubocop/cop/offense.rb#L143C35-L143C46) which is a string. I suspect this issue coming for this mismatch. 👍
